### PR TITLE
Replace project-specific scylladb_markdown extension with sphinx-scylladb-markdown extension

### PR DESCRIPTION
**Pull request by dgarcia360**
Related issue https://github.com/scylladb/sphinx-scylladb-theme/issues/168

Replaces the project-specific Markdown extension with the sphinx-scylladb-markdown extension from [sphinx-scylladb-theme](https://github.com/scylladb/sphinx-scylladb-theme/tree/master/extensions/sphinx-scylladb-markdown), available on PyPI for all projects.

## How to test

Docs should be built and rendered without errors.

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -64,7 +64,6 @@
 # -- Options for markdown extension
 scylladb_markdown_enable = True
 scylladb_markdown_recommonmark_versions = ['v1.9', 'v1.10']
-myst_heading_anchors = 6
```
Comment (rzetelskik) [inline docs/source/conf.py]: This line is added in the first commit and then removed in the second. Please squash the commits.

**Review by rzetelskik — CHANGES_REQUESTED** [review]
@dgarcia360 thanks, left one comment. Could you also please update the commit name to align with our repository? Something along the lines of `Replace project-specific scylladb_markdown extension with sphinx-scylladb-markdown extension` (or at least remove the `docs:` prefix).

**Comment by dgarcia360** [comment]
> Could you also please update the commit name to align with our repository?

Done!

**Review by tnozicka — APPROVED** [review]
/approve
/lgtm

thanks!

**Pull request merged**